src/wp-includes/class-wp.php

```diff
@@ -367,6 +367,99 @@ public function parse_request( $extra_query_vars = '' ) {
 			}
 		}
 
+		// Prevent invalid date queries.
```

Comment:
**suggestion (if-minor):** it would be nice to move this validation to a separate class, wdyt ?